manager/catalog_manager.go

```diff
@@ -328,17 +335,40 @@ func ReadTemplateVersion(catalogID string, templateID string, versionID string)
 func GetNewTemplateVersions(path string) (model.Template, bool) {
 	templateMetadata := model.Template{}
 
+	log.Debugf("Over here path : %v\n", path)
 	//find the base template metadata name
 	tokens := strings.Split(path, "/")
 	catalogID := tokens[0]
 	parentPath := tokens[1]
 	cVersion := tokens[2]
 
 	//refresh the catalog and sync any new changes
+
 	cat := CatalogsCollection[catalogID]
-	//cat.refreshCatalog()
+	cat.refreshCatalog()
+
+	rancherComposePathCurrent := CatalogRootDir + catalogID + "/templates/" + parentPath + "/" + cVersion
+	log.Debugf("Exact path : %s\n", rancherComposePathCurrent)
+
+	composeBytes, err := readFile(rancherComposePathCurrent, "rancher-compose.yml")
+	if err != nil {
+		return templateMetadata, false
+	}
+
+	//read the questions section
+	RC := make(map[string]model.RancherCompose)
+	err = yaml.Unmarshal(*composeBytes, &RC)
+	if err != nil {
+		log.Errorf("Error unmarshalling %s under template: %s, error: %v", "rancher-compose.yml", rancherComposePathCurrent, err)
+		return templateMetadata, false
+	}
+	current_version := RC[".catalog"].Version
+	if strings.Count(current_version, ".") < 2 {
```

Comment:
This should go in getSemVersion so it's not repeated.

Needs to be a little more robust to work for an input like `v1.2-pre1` -> `v1.2.0-pre1`, not `v1.2-pre1.0`